### ISSUE TITLE
Use without Google/AWS account

### ISSUE THREAD

**Issue by obfuscoder** [rkusa/DATIS#24]
Both cloud text-to-speech services require to add payment information (credit card) although it won't be used ever due to the small amount of characters to be converted to speech each month. Is there a way to let DATIS just generate the text form the DCS mission  so that we can use a text-to-speech engine manually by copy&pasting the text and placing the generated audio file somewhere side by side with the miz file?

**Comment by obfuscoder** [comment]
The best approach would be that the text-to-speech engine would be pluggable. The input is the generated text by DATIS, the output would be an audio file (mp3 or otherwise) which DATIS can then integrate. There are pretty decent tools which can be used. On Windows you have the Microsoft Narrator built-in for instance. A lot of free tools with GUI are available which can be interacted with by an AutoIt script to paste the text and create the audio file.

**Comment by obfuscoder** [comment]
Just implemented a small .Net console application using the Microsoft speech engine to generate an audio file from a text file. Can be done in 3 lines of code:

```
SpeechSynthesizer synth = new SpeechSynthesizer();
synth.SetOutputToWaveFile("output.wav");
synth.Speak(File.ReadAllText("input.txt"));
```

The result is great and no external service required at all.

Only .Net 3.0 is required which is shipped with Windows Vista and later.

**Comment by rkusa** [comment]
> Is there a way to let DATIS just generate the text form the DCS mission so that we can use a text-to-speech engine manually by copy&pasting the text and placing the generated audio file somewhere side by side with the miz file?

There might be a workaround. Insert some random characters into the gcloud key input (doesn't need to be a valid key). While when you open a mission the TTS will error, the generated report should already be saved into `DCS.openbeta\Logs\atis-reports.json`. You can copy it from there and find a tool to generate a sound file from it you then add into the mission (AFAIC, there are triggers to play audio files at a certain frequency).

> Just implemented a small .Net console application using the Microsoft speech engine to generate an audio file from a text file.

With the upcoming version (currently `1.0.0-alpha.1`), where we've added AWS as an additional TTS provider, we did the ground work to add additional TTS engines. I already have it on my list to look into calling the MS TTS engine you've mentioned from Rust.

**Comment by rkusa** [comment]
Since I was playing around with using Windows' speech recognition anyway, interating its TTS wasn't a big deal.

Initially, I was planning to directly call to the winapi from Rust, but I've ended up with a way simpler approach for now, which is basically almost your suggest three liner compiled into a seperate executable which is executed from Rust. https://github.com/rkusa/DATIS/tree/master/win/win-tts

Released as https://github.com/rkusa/DATIS/releases/tag/1.0.0-alpha.2

**Comment by obfuscoder** [comment]
Great work. The only caveat is that it will create a wav file which is getting quite big. Might be worth using libogg to convert the wav into an ogg file.

**Comment by obfuscoder** [comment]
One additional note: Windows TTS is shipping with 3 voices - 2 male and one female. Would be nice to add and option to have a choice there. We would love to use the female voice for ATIS ;-)

**Comment by rkusa** [comment]
> Great work. The only caveat is that it will create a wav file which is getting quite big. Might be worth using libogg to convert the wav into an ogg file.

I skipped the step of creating a wav file. Instead, it writes the wav to stdout, which is received by the calling Rust program where it is converted to ogg/opus.

> One additional note: Windows TTS is shipping with 3 voices - 2 male and one female. Would be nice to add and option to have a choice there. We would love to use the female voice for ATIS ;-)

I saw that, but postponed adding this option until a later point in time. Btw, I only have two engl. voices pre-installed not three

**Comment by rkusa** [comment]
You could, in the meantime, update https://github.com/rkusa/DATIS/blob/master/win/win-tts/Program.cs with the voice you want, build it, and put the `win-tts.exe` into `Saved Games\DCS.openbeta\Mods\tech\DATIS\bin`

**Comment by obfuscoder** [comment]
Yeah good point. I will extend it and might send a PR your way.

**Comment by rkusa** [comment]
@obfuscoder FYI, this version released earlier today (1.0.0-alpha.2) might crash DCS when used on a server - I am working on fixing that ...
